lib/active_merchant/billing/gateways/usa_epay_transaction.rb

```diff
@@ -178,7 +178,8 @@ def add_credit_card(post, credit_card)
           post[:card]   = credit_card.number
           post[:cvv2]   = credit_card.verification_value if credit_card.verification_value?
           post[:expir]  = expdate(credit_card)
-          post[:name]   = credit_card.name
+          post[:name]   = credit_card.name unless credit_card.name.blank?
+          post[:cardpresent] = true if credit_card.manual_entry
```

Comment:
Thinking about it more - I think what you have here reads clearer right now, especially if this isn't required for all transactions.